src/main/scala/mastermind/Mastermind.scala

```diff
@@ -10,22 +10,31 @@ import scala.io.StdIn.readLine
 type Color = Char
 type Board = String
 val validColors = List('B', 'Y', 'R', 'G')
+val random = new Random()
 
 /** Get a random color from the list of valid colors */
 def getRandomColor(): Color =
-  ???
+  validColors.apply(random.nextInt(4))
+
 
 /** Given four colors, make a board from them */
 def makeBoardFromColors(c1: Color, c2: Color, c3: Color, c4: Color): Board =
-  ???
+  List(c1, c2, c3, c4).mkString
 
 /** Create a random board */
 def getRandomBoard(): Board =
-  ???
+  makeBoardFromColors(getRandomColor(),getRandomColor(), getRandomColor(), getRandomColor())
 
 /** Play one round of the game */
-def playRound(board: Board): (Int, Int) =
-  ???
+def playRound(board: Board): (Int, Int) = 
```

Comment:
I like your use of toUpperCase().toCharArray() in order to make sure users inputs are all uppercase, also you could use .charAt(0) instead of to char array to further simplify your code and make it more readable